### PR TITLE
W-15609911: Provide a common parent pom with pluginManangement of common

### PR DESCRIPTION
**Pull request by elrodro83**
* Extract common release config
* Cleanup
* Address pending technical debt

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Review by anosenzo — APPROVED** [review]

File: modules/artifact/pom.xml [hunk 1]
```diff
@@ -31,26 +31,28 @@
...
-                            <includes>
-                                <include>org/mule/runtime/module/artifact/api/classloader/test/**</include>
-                                <include>org/mule/runtime/module/artifact/internal/util/test/**</include>
-                            </includes>
-                        </configuration>
-                    </execution>
-                </executions>                
-            </plugin>
-        </plugins>
+        <pluginManagement>
```
Comment (anosenzo) [inline modules/artifact/pom.xml:34]: Why this change was necessary?

Comment (elrodro83) [reply modules/artifact/pom.xml:34]: the plugin is already declared to run, this is jsut for setting some config without overridicng the stadard configuraton from the parent
